### PR TITLE
Use checked-in platform manifest in 3.1.2

### PR DESCRIPTION
**Pull request by wtgodbe**
I generated this platform manifest based off of the 3.1.0 runtime pack. Checking it into the repo is sufficient for 3.1.2, since there is no "known good" platform manifest for 3.1.x that we can harvest and insert into the package. Replaces https://github.com/dotnet/aspnetcore/pull/17957. Copied from there:

> https://github.com/aspnet/AspNetCore/pull/14538 changed `PlatformManifest.txt` in the Targeting pack to list the assemblies in the targeting pack, rather than to list the assemblies in the runtime pack. At the time, we thought the former was the correct behavior, but in reality the latter is correct. The platform manifest exists to inform an app, while building against the ref pack assemblies, what will be available in the shared framework at runtime. We changed this behavior between 3.0.0 and 3.0.1, which broke customers who had transitive 2.x dependencies in their 3.0 apps (specifically `aspnetcorev2_inprocess.dll`).

> We have a somewhat ugly workaround for this and are working on finding a better one.

I'll open an issue to track coming up with a better solution in case we need to service the targeting pack in the future - ideally a solution would be general across all repos that generate a targeting pack

CC @nguerrera @dagood @ericstj @Pilchie @JunTaoLuo @jkotalik @dougbu

### CODE REVIEW

File: Directory.Build.props [hunk 1]
```diff
@@ -84,11 +84,11 @@
...
     <TargetingPackInstallerBaseName>aspnetcore-targeting-pack</TargetingPackInstallerBaseName>
 
-    <!-- Produce targeting pack installers/packages once per major.minor except in extraordinary cases i.e. 3.0.1. -->
-    <!-- We can remove the 3.0.1 line from any branch other than release/3.0 and from here after 3.0.1 is released. -->
+    <!-- Produce targeting pack installers/packages once per major.minor except in extraordinary cases i.e. 3.1.2. -->
+    <!-- We can remove the 3.1.2 line from any branch other than release/3.1 and from here after 3.1.2 is released. -->
     <IsTargetingPackBuilding Condition=" '$(DotNetBuildFromSource)' == 'true' ">false</IsTargetingPackBuilding>
     <IsTargetingPackBuilding
-        Condition=" '$(IsTargetingPackBuilding)' == '' AND '$(VersionPrefix)' == '3.0.1' ">true</IsTargetingPackBuilding>
+        Condition=" '$(IsTargetingPackBuilding)' == '' AND '$(VersionPrefix)' == '3.1.2' ">true</IsTargetingPackBuilding>
```
Comment (JunTaoLuo) [inline Directory.Build.props]: FYI I already made these changes in https://github.com/dotnet/aspnetcore/pull/18151/files

File: eng/Versions.props [hunk 2]
```diff
@@ -33,7 +33,7 @@
     <!-- TargetingPackVersionPrefix is used by projects, like .deb and .rpm, which use slightly different version formats. -->
     <TargetingPackVersionPrefix>$(VersionPrefix)</TargetingPackVersionPrefix>
     <!-- Targeting packs do not produce patch versions in servicing builds. No API changes are allowed in patches. -->
-    <TargetingPackVersionPrefix Condition="'$(IsServicingBuild)' == 'true'">$(AspNetCoreMajorVersion).$(AspNetCoreMinorVersion).0</TargetingPackVersionPrefix>
+    <TargetingPackVersionPrefix Condition="'$(IsTargetingPackBuilding)' != 'true'">$(AspNetCoreMajorVersion).$(AspNetCoreMinorVersion).2</TargetingPackVersionPrefix>
```
Comment (JunTaoLuo) [inline eng/Versions.props]: 👍

**Review by Pilchie — APPROVED** [review]
This looks good to me based on the conversations I've had so far with @ericstj and @nguerrera

File: src/Framework/ref/Microsoft.AspNetCore.App.Ref.csproj [hunk 3]
```diff
@@ -53,7 +53,7 @@ This package is an internal implementation of the .NET Core SDK and is not meant
     <!-- Platform manifest and package override metatdata -->
     <ReferencePackSharedFxVersion>$(AspNetCoreMajorVersion).$(AspNetCoreMinorVersion).0</ReferencePackSharedFxVersion>
     <ReferencePackSharedFxVersion Condition="'$(VersionSuffix)' != ''">$(ReferencePackSharedFxVersion)-$(VersionSuffix)</ReferencePackSharedFxVersion>
-    <ReferencePlatformManifestOutputPath>$(ArtifactsObjDir)ref\PlatformManifest.txt</ReferencePlatformManifestOutputPath>
+    <ReferencePlatformManifestPath>$(RepoRoot)eng\PlatformManifest.txt</ReferencePlatformManifestPath>
```
Comment (dougbu) [inline src/Framework/ref/Microsoft.AspNetCore.App.Ref.csproj]: In non-servicing builds, this should copy from the just-built runtime pack. That way, we can merge this change into 'master'.

**Review by dougbu — CHANGES_REQUESTED** [review]

**Comment by wtgodbe** [comment]
This was approved for 3.1.2 in https://github.com/dotnet/aspnetcore/pull/17957

File: src/Framework/ref/Microsoft.AspNetCore.App.Ref.csproj [hunk 4]
```diff
@@ -53,7 +53,8 @@ This package is an internal implementation of the .NET Core SDK and is not meant
     <!-- Platform manifest and package override metatdata -->
     <ReferencePackSharedFxVersion>$(AspNetCoreMajorVersion).$(AspNetCoreMinorVersion).0</ReferencePackSharedFxVersion>
     <ReferencePackSharedFxVersion Condition="'$(VersionSuffix)' != ''">$(ReferencePackSharedFxVersion)-$(VersionSuffix)</ReferencePackSharedFxVersion>
-    <ReferencePlatformManifestOutputPath>$(ArtifactsObjDir)ref\PlatformManifest.txt</ReferencePlatformManifestOutputPath>
+    <ReferencePlatformManifestPath Condition="'$(IsServicingBuild)' != 'true'">$(PlatformManifestOutputPath)</ReferencePlatformManifestPath>
```
Comment (wtgodbe) [inline src/Framework/ref/Microsoft.AspNetCore.App.Ref.csproj]: @dougbu does this resolve your comment?

Comment (dougbu) [reply src/Framework/ref/Microsoft.AspNetCore.App.Ref.csproj]: Yup

Comment (dougbu) [inline src/Framework/ref/Microsoft.AspNetCore.App.Ref.csproj]: nit:
```suggestion
    <ReferencePlatformManifestPath>$(PlatformManifestOutputPath)</ReferencePlatformManifestPath>
```

**Review by dougbu — APPROVED** [review]
Looks good but please do a fake non-servicing build after cleaning your repo e.g. port to 'master'. That'll confirm build ordering doesn't mess up the changes to the App.Ref. (App.Runtime depends on App.Ref, not the other way 'round. But, I'm not sure about when the specific packaging targets run.)

**Pull request merged**